### DISCUSSION TITLE
Does mdbx_strerror_r's return value require free-ing?

### DISCUSSION THREAD

**Discussion by kozross** [erthink/libmdbx#359] [Q&A] [answered]
After something like:

```c
// assume we define appropriate variables
const char * error_text = mdbx_strerror_r(errnum, buffer, 1024);
```

Do I need to call ``free`` on ``error_text``?

**Comment by erthink** [comment, marked as answer]
You should manage the `buffer` (and free it in case it was allocated), but not the `error_text` (which cold be pointed to const string or to any byte inside the `buffer`).

Like the GNU-specific `strerror_r()`, the `mdbx_strerror_r()` returns either a pointer to a string that the function stores in given `buffer`, or a pointer to  some (immutable) static string (in which case `buffer` is unused).
